Content.Client/Paint/PaintVisualizerSystem.cs

```diff
@@ -0,0 +1,105 @@
+using System.Linq;
+using Robust.Client.GameObjects;
+using static Robust.Client.GameObjects.SpriteComponent;
+using Content.Shared.Clothing;
+using Content.Shared.Hands;
+using Content.Shared.Paint;
+
+namespace Content.Client.Paint
+{
+    public sealed class PaintedVisualizerSystem : VisualizerSystem<PaintedComponent>
+    {
+        /// <summary>
+        /// Visualizer for Paint which applies a shader and colors the entity.
+        /// </summary>
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<PaintedComponent, HeldVisualsUpdatedEvent>(OnHeldVisualsUpdated);
+            SubscribeLocalEvent<PaintedComponent, ComponentShutdown>(OnShutdown);
+            SubscribeLocalEvent<PaintedComponent, EquipmentVisualsUpdatedEvent>(OnEquipmentVisualsUpdated);
+        }
+
+        // Applies the shader and color to all sprite layers for the entity.
+        protected override void OnAppearanceChange(EntityUid uid, PaintedComponent component, ref AppearanceChangeEvent args)
+        {
+            if (args.Sprite == null)
+                return;
+
+            if (!TryComp(uid, out SpriteComponent? sprite))
+                return;
+
```

Comment:
If the args sprite is null then trycomp is also going to be null so that's unnecessary.

Reply:
removed trycomp here.